mlrun/track/tracker_manager.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Type, Union
+
+from mlrun.execution import MLClientCtx
+from mlrun.utils import logger
+
+from .tracker import Tracker
+
+
+class TrackerManager:
+    """
+    A class for handling multiple `Tracker` instances during a run. The manager is a singleton class and should be
+    retrieved using the `mlrun.track.get_trackers_manager` function.
+    """
+
+    def __init__(self, _stale: bool = False):
+        """
+        Initialize a new empty tracker manager.
+
+        :param _stale: An inner attribute to init a trackers manager in a specific staleness state.
+        """
+        self._trackers: List[Type[Tracker]] = []
+        self._stale = _stale
+
+    def add_tracker(self, tracker: Type[Tracker]):
+        """
+        Adds a Tracker to trackers list.
+
+        :param tracker: The tracker class to add
+        """
+        self._trackers.append(tracker)
+        logger.debug("Added tracker", tracker=tracker)
+
+    def pre_run(self, context: MLClientCtx):
+        """
+        Goes over all the managed trackers and calls their `pre_run` method.
+
+        :param context: Active mlrun context.
+        """
+        for tracker in self._trackers:
+            tracker.pre_run(context=context)
```

Comment:
if once fails, do you want to continue or bail?

Reply:
imo bail. they are configured to run by the user, if it fails the user will probably don't want the function to run. wdyt?

Reply:
agree. how about post run? these would fail the entire run operation.

Reply:
hmm, I think these should not fail. 

@ZeevRispler Please wrap each call in try except and log as warning when a tracker raised an exception 💪

Reply:
done